### PR TITLE
feat(marketing): add contentful experiences

### PR DESCRIPTION
**Pull request by stephenliang**
This PR initalizes Contenful Experiences in the marketing app. The following [guide](https://www.contentful.com/developers/docs/experiences/set-up-experiences-sdk/) was used.

The `Stub` and `StubSection` components are also updated to be registered custom components in Contentful and has been tested on Contentful Studio.

The process of connecting `marketing` to `dsco` revealed a number of integration issues, which this PR resolves:

1. Initially, the `dsco` package would automatically create entrypoint export entries in package.json via `clean-package`. However, this does not work due to `dsco` being linked (it is not being published and pulled down) and therefore dynamic entrypoints are not available.
2. The `dsco` package included a tsup banner for `use client;`, however this is not needed as each design system component should be SSR compatiable.
3. `import-x/no-unresolved` does not appear to work with our monorepo setup at this time, so a separate task was created to investigate and fix this.

### CODE REVIEW

File: frontend/apps/marketing/.env.example
Comment (kelbyhawn) [inline frontend/apps/marketing/.env.example:10]: Process Q for after this is merged: do we need to fill these in locally with a `.env` copy of this file? Similar to what we've done before with the `APPLITOOLS_API_KEY` token?

File: frontend/apps/marketing/src/contentful/register-custom-components.ts
Comment (kelbyhawn) [inline frontend/apps/marketing/src/contentful/register-custom-components.ts]: Does each new component need to be manually added to this file when we want it to be available on Studio/Experiences?

File: frontend/packages/dsco/package.json
Comment (kelbyhawn) [inline frontend/packages/dsco/package.json:20]: Do components get added here automatically?

**Review by kelbyhawn — APPROVED** [review]
I have a few misc questions, but I think this looks good based on the docs link you shared! Others should take a look though to check the `marketing` to `dsco` integration issues.

**Comment by bencodeorg** [comment]
Could we update the readme in the `marketing` directory to describe how to get integrated with Contentful Studio aka Experiences? FWIW I was able to get your hello world page up and running 🎉 , but it'd be nice to have a bit of documentation for folks with less Contentful experience.

**Comment by bencodeorg** [comment]
> however this is not needed as each design system component should be SSR compatiable.

That's not true of the existing component library (right?) Are we planning on adding `use client` to the top of each component? Or actually making them SSR-compatible during the migration?

File: frontend/apps/marketing/.env.example
Comment (stephenliang) [reply frontend/apps/marketing/.env.example:10]: Correct - and I'll update the README to include this.

File: frontend/apps/marketing/src/contentful/register-custom-components.ts
Comment (stephenliang) [reply frontend/apps/marketing/src/contentful/register-custom-components.ts]: Yes, only if we want it to show up Contentful Studio Experiences. Updated the comment to clarify

File: frontend/packages/dsco/package.json
Comment (stephenliang) [reply frontend/packages/dsco/package.json:20]: No, I wanted to do that originally with the `clean-package` approach but had to delete it due to the way we're doing yarn workspace linking (the approach I was going with only worked in the case we did a publish/pull model but it appears we are not going to be using that model internally). This means each time we add a new component, we have to add an export here. 

If this becomes problematic, we will need to see if there's a way to update this programmatically.

Let me add a section on README on how to create new components to clarify this.

**Comment by stephenliang** [comment]
> Could we update the readme in the `marketing` directory to describe how to get integrated with Contentful Studio aka Experiences? FWIW I was able to get your hello world page up and running 🎉 , but it'd be nice to have a bit of documentation for folks with less Contentful experience.

I added instructions on the README to copy the `.env.example` file, which has instructions on how to get the value for each entry.

**Comment by stephenliang** [comment]
> > however this is not needed as each design system component should be SSR compatiable.
> 
> That's not true of the existing component library (right?) Are we planning on adding `use client` to the top of each component? Or actually making them SSR-compatible during the migration?

The existing component library (`apps/src/componentLibrary`) is fully client-side rendered, but since we're looking to do SEO, we do need the component library to also be SSR capable.

File: frontend/apps/marketing/README.md
Comment (bencodeorg) [inline frontend/apps/marketing/README.md:25]: Should we also include some info about how to use this on the Contentful side (eg, setting the appropriate "Preview Platform" like shown in [this thread?](https://codedotorg.slack.com/archives/C07UW4ED66Q/p1733441644442949?thread_ts=1733440885.431029&cid=C07UW4ED66Q))

Also, longer term, but would we eventually include some guidance to use the `dev` Contentful environment?

Comment (stephenliang) [reply frontend/apps/marketing/README.md:25]: I'm thinking once we have a good rhythm we can go back and create the correct docs. Right now things are in flight with the temporary AWS setup and I'm afraid we'll leave some outdated information. I have a task to slot that in perhaps in mid to late January to back fill all documentation since I'm guessing we'll be getting in the rhythm of creating and integrating by then.

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**
